### ISSUE TITLE
[Improvement] Searchspace's type "randint" is better to support lower bound

### ISSUE THREAD

**Issue by PurityFan** [microsoft/nni#902]
**Short summary about the issue/question**:
Currently, the search space in NNI only supports setting upper bound. 
If I want to set {batch_size: [32, 256]}, I can't choose randint, I have to use:
"batch_size":{"_type":"quniform", "_value":[32, 256, 1]}
But this will return a float, and you need to convert it to int in your code.

Suggest give the chose to the user, "randint" support lower bound, like 
{“_type”:”randint”,”_value”:[low, high]}
If you want to use the original one, you can set {“_type”:”randint”,”_value”:[0, high]}
This can enhance its generalization

**nni Environment**:
- nni version: 0.5.2

**Comment by xuehui1991** [comment]
Good suggestion!

**Comment by squirrelsc** [comment]
I need this also, for example in random forest, number of leaves in a node can be a range from 10 to 100, but I don't want it randomized to 0, 1, 2.

**Comment by xuehui1991** [comment]
Actually, we might make a mistake about the search space type of "randint".  

"randint" Which means the variable value is a random integer in the range [0, upper). The semantics of this distribution is that there is **no more correlation in the loss function** between nearby integer values, as compared with more distant integer values.   

So I think "randint" is more like "choice". @squirrelsc @PurityFan  in your scenario, we might use quniform to instead.

**Comment by PurityFan** [comment]
But the hyperparameter gets from quniform get is a **float**. 
So if user have to get a rand int from `[32, 256]`, use `arg['batch_size'] + 32`(by randint) or `int(arg['batch_size'])`(by quniform) is bothersome

**Comment by squirrelsc** [comment]
other concern is the choice is not good for big range like [32, 256), it means more than 200 choices need to be written.

**Comment by silverlining21** [comment]
> But the hyperparameter gets from quniform get is a **float**.
> So if user have to get a rand int from `[32, 256]`, use `arg['batch_size'] + 32`(by randint) or `int(arg['batch_size'])`(by quniform) may be bothersome

cant be more agree~
